cmd/charm/charmcmd/publish.go

```diff
@@ -116,6 +116,31 @@ func (c *publishCommand) Run(ctxt *cmd.Context) error {
 	}
 	fmt.Fprintln(ctxt.Stdout, "url:", c.id)
 	fmt.Fprintln(ctxt.Stdout, "channel:", c.channel)
+	if "stable" == c.channel {
```

Comment:
No. That defensive style makes sense coming from a C background, which allows assignments in an if statement if you forgot an `=`. It's neither necessary nor idiomatic to do this in Go. The compiler will prevent unintentional assignment in an if expression.